common/v2/features/BuyAssets/BuyAssetsForm.tsx

```diff
@@ -0,0 +1,199 @@
+    asset: ethAsset
+  };
+
+  const BuyFormSchema = Yup.object().shape({
+    account: Yup.object().required(translateRaw('REQUIRED')),
+    asset: Yup.object().required(translateRaw('REQUIRED'))
+  });
+
+  const filteredAssets = assets.filter(({ uuid }) => MOONPAY_ASSET_UUIDS.includes(uuid));
+
+  const handleSubmission = (values: IBuyFormState, typeOfSubmission: SubmissionType) => {
+    if (typeOfSubmission === SubmissionType.SEND_TO_SELF) {
+      const urlQuery = `${MOONPAY_API_QUERYSTRING}&currencyCode=${values.asset.ticker}&walletAddress=${values.account.address}`;
+
+      MoonpaySignerService.instance
+        .signUrlQuery(urlQuery)
+        .then((signature: any) => {
+          const redirectQueryParams = `?currencyCode=${values.asset.ticker}&walletAddress=${
+            values.account.address
+          }&signature=${encodeURIComponent(signature)}`;
+          window.open(`${BUY_MYCRYPTO_WEBSITE}${redirectQueryParams}`, '_blank');
+        })
+        .catch(err => {
+          console.debug('err detected: ', err);
+          window.open(`${BUY_MYCRYPTO_WEBSITE}`, '_blank');
+        });
+    } else {
+      window.open(`${BUY_MYCRYPTO_WEBSITE}`, '_blank');
+    }
+  };
+
+  const goBack = () => history.push(ROUTE_PATHS.DASHBOARD.path);
+
+  return (
+    <ContentPanel
+      onBack={goBack}
+      backBtnText={translateRaw('DASHBOARD')}
+      heading={translateRaw('DASHBOARD_ACTIONS_BUY_TITLE')}
+    >
+      <Formik
+        initialValues={initialFormikValues}
+        validationSchema={BuyFormSchema}
+        onSubmit={vals => handleSubmission(vals, SubmissionType.SEND_TO_SELF)}
+        render={({ values, errors, touched, setFieldValue }) => {
+          const relevantAccounts = accounts.filter(account => {
+            return values.asset && values.asset.networkId
+              ? isAccountInNetwork(account, values.asset.networkId)
+              : true;
+          });
+
+          return (
+            <Form>
+              <FormFieldItem>
+                <FormFieldLabel htmlFor="account">{translate('X_RECIPIENT')}</FormFieldLabel>
+                <Field
+                  name="account"
+                  value={values.account}
+                  component={({ field }: FieldProps) => (
+                    <AccountDropdown
+                      name={field.name}
+                      value={field.value}
+                      accounts={relevantAccounts}
+                      onSelect={(option: IAccount) => {
+                        setFieldValue('account', option); //if this gets deleted, it no longer shows as selected on interface, would like to set only object keys that are needed instead of full object
+                      }}
+                    />
+                  )}
+                />
+                {errors && errors.account && touched && touched.account ? (
+                  <InlineMessage className="SendAssetsForm-errors">{errors.asset}</InlineMessage>
+                ) : null}
```

Comment:
Ternary operator can be removed in this case. The same for the lines 170-172.
```suggestion
{errors && errors.account && touched && touched.account && (
  <InlineMessage className="SendAssetsForm-errors">{errors.asset}</InlineMessage>
)}
```